girder/api/docs.py

```diff
@@ -112,13 +112,25 @@ def removeRouteDocs(resource, route, method, info, handler):
                 del routes[resource]
 
 
-def addModel(name, model):
+def addModel(resource, name, model):
     """
-    This is called to add a model to the swagger documentation.
+    Add a model to the Swagger documentation.
 
+    :param resource: The type of existing resource to add the model to.
+    :param resource: str
     :param name: The name of the model.
     :type name: str
     :param model: The model to add.
     :type model: dict
```

Comment:
That is a possibility that I considered, but I'm worried that it will cause a lot more long-term pain. This function has no documentation that's built, and it's located in a fairly obscure place. @zachmullen and I both initially thought it was unused. I'm hoping that @cjh1 is the only person who currently uses this function (no surprising since he wrote it), and with his cooperation, we can just slip the API change in under the radar. The longer we wait, and the longer we tolerate the existing behavior, the more likely it is that others will start using the function, and then we're really stuck.

The semantics of a default argument are generally such that it behaves sensibly and sanely by default, while allowing you to optionally specify additional customization. An implementation that adds a model to every current resource is probably broken behavior (given Girder's semantics), and adding it to every future-defined resource is...yikes. I think having this behavior be the default is asking for trouble.

If the consensus is that we really can't slip in this API change, I'd propose creating a new function with the proper arguments (unfortunately, the best, most appropriate name is already taken), while slapping deprecation warnings all over this existing one. Then the behavior of the existing one can do whatever we want (including preserving the current behavior), but it's unlikely that new developers will start to rely on it. We can then safely remove the existing one in Girder 2.0 without any guilt.